### PR TITLE
Add kill submission resource to kubejobs plugin

### PR DESCRIPTION
**Pull request by javanlacerda**
- created the rest endpoint to submit requests to kill a job
- created the call to k8s api in the plugin
- created in k8s the functionality that kill all resources referents to job

### CODE REVIEW

File: broker/plugins/base.py [hunk 1]
```diff
@@ -128,6 +128,10 @@ class ApplicationExecutor(object):
     @required
     def get_application_state(self):
         pass
+    
+    @required
+    def kill_himself(self):
```
Comment (GabrielSVinha) [inline broker/plugins/base.py]: A better name for this function would be "clean" or "remove_submission".

File: broker/plugins/base.py [hunk 2]
```diff
@@ -149,6 +153,9 @@ def __init__(self):
 
     def get_application_state(self):
         pass
+    
+    def kill_himself(self):
```
Comment (GabrielSVinha) [inline broker/plugins/base.py]: Same as above.

File: broker/plugins/kubejobs/plugin.py [hunk 3]
```diff
@@ -119,6 +119,10 @@ def get_application_state(self):
     def update_application_state(self, state):
         self.status = state
 
+    def kill_himself(self):
```
Comment (GabrielSVinha) [inline broker/plugins/kubejobs/plugin.py]: Same as above.

**Comment by ricardoas** [comment]
It seems ok... I agree with @GabrielSVinha about the naming conventions.

Also, is it ok to use rebase instead of merging when updating the next time? I think it is a better approach... wdyt?

**Pull request closed without merging**